src/ifixit2zim/utils.py

```diff
@@ -67,15 +76,20 @@ def only_path_of(self, url: str):
         """normalized path part of an url"""
         return self.normalize_ident(urllib.parse.urlparse(url).path)
 
+    @backoff.on_exception(
+        backoff.expo,
+        requests.exceptions.HTTPError,
+        max_time=16,
+        on_backoff=backoff_hdlr,
+        giveup=fatal_code,
+    )
     def fetch(self, path: str, **params) -> tuple[str, list[str]]:
         """(source text, actual_paths) of a path from source website
 
         actual_paths is amn ordered list of paths that were traversed to get to content.
         Without redirection, it should be a single path, equal to request
         Final, target path is always last"""
-        session = requests.Session()
-        session.mount("http", _get_retry_adapter(10))  # tied to http and https
-        resp = session.get(self.get_url(path, **params), params=params)
+        resp = requests.get(self.get_url(path, **params), params=params, timeout=10)
```

Comment:
Please extract the `10` magic number in a constant so that it is shared between both methods and more readable.

Or maybe we should extract this logic around retries + call to requests.get + raise_for_status in a shared method so that there is less duplication ? Both options are ok for me.